components/homme/src/share/vertremap_base.F90

```diff
@@ -610,26 +610,25 @@ subroutine remap_Q_ppm(Qdp,nx,qsize,dp1,dp2)
           masso(k+1) = masso(k) + ao(k) !Accumulate the old mass. This will simplify the remapping
           ao(k) = ao(k) / dpo(k)        !Divide out the old grid spacing because we want the tracer mixing ratio, not mass.
         enddo
-        !Fill in ghost values. Ignored if vert_remap_q_alg == 2
-        if (vert_remap_q_alg >= 1 .and. vert_remap_q_alg <= 3) then
+        !Fill in ghost values.
+        if (remap_alg==10) then
```

Comment:
I recommend a slightly cleaner approach here to avoid code duplication:
```F90
logical :: limit_linear_extrap
...
if (remap_alg == 10 .or. remap_alg == 11) then
  limit_linear_extrap = remap_alg == 10
  if (limit_linear_extrap) then
    ext(1) = minval(ao(1:nlev))
    ext(2) = maxval(ao(1:nlev))
  end if
  call linextrap(dpo(2), dpo(1), dpo(0), dpo(-1), ao(2), ao(1), ao(0), ao(-1), limit_linear_extrap, ext(1), ext(2))
  call linextrap(dpo(nlev-1), dpo(nlev), dpo(nlev+1), dpo(nlev+2),&
      ao(nlev-1), ao(nlev), ao(nlev+1), ao(nlev+2), limit_linear_extrap, ext(1), ext(2))
end if
```
Then make the third-to-last argument a `logical` in `linextrap`.

Reply:
good idea - now done.